### PR TITLE
NEW : element time integration code + sql

### PR DESCRIPTION
**Pull request by atm-gauthier**

### CODE REVIEW

**Comment by atm-gauthier** [comment]
I got the same travis error than in pr https://github.com/Dolibarr/dolibarr/pull/24116, which doesn't seem to deal with what i did

**Comment by frederic34** [comment]
you need to have your sql migration script in this PR because invoice delete is using the new table

File: htdocs/compta/facture/class/facture.class.php [hunk 1]
```diff
@@ -2690,7 +2690,7 @@ public function delete($user, $notrigger = 0, $idwarehouse = -1)
 				return -5;
 			}
 
-			$sql = 'UPDATE '.MAIN_DB_PREFIX.'projet_task_time';
+			$sql = 'UPDATE '.MAIN_DB_PREFIX.'element_time';
 			$sql .= ' SET invoice_id = NULL, invoice_line_id = NULL';
 			$sql .= ' WHERE invoice_id = '.((int) $rowid);
```
Comment (frederic34) [inline htdocs/compta/facture/class/facture.class.php:2695]: without sql migration this request fails

Comment (atm-gauthier) [reply htdocs/compta/facture/class/facture.class.php:2695]: Hi @frederic34 thank you very much for your help, i didn't thought about that ;)
I'm going to merge these 2 PR for this case

**Comment by frederic34** [comment]
https://github.com/Dolibarr/dolibarr/pull/24165

File: htdocs/install/mysql/migration/17.0.0-18.0.0.sql [hunk 2]
```diff
@@ -100,3 +100,19 @@ ALTER TABLE llx_recruitment_recruitmentcandidature ADD COLUMN fk_user integer;
...
+-- Element time
+ALTER TABLE llx_projet_task_time RENAME TO llx_element_time;
+ALTER TABLE llx_element_time CHANGE COLUMN fk_task fk_element integer NOT NULL;
+ALTER TABLE llx_element_time CHANGE COLUMN task_date element_date date;
+ALTER TABLE llx_element_time CHANGE COLUMN task_datehour element_datehour datetime;
+ALTER TABLE llx_element_time CHANGE COLUMN task_date_withhour element_date_withhour integer;
+ALTER TABLE llx_element_time CHANGE COLUMN task_duration element_duration double;
+ALTER TABLE llx_element_time ADD COLUMN elementtype varchar(32) NOT NULL DEFAULT 'task' AFTER fk_element;
+
+DROP INDEX idx_projet_task_time_task on llx_element_time;
```
Comment (eldy) [inline htdocs/install/mysql/migration/17.0.0-18.0.0.sql:112]: To drop an index, you must use 2 lines (one for mysql, one for postgresql). See examples on top of file.
So
-- VMYSQL4.1 DROP INDEX nomindex on llx_table;
-- VPGSQL8.2 DROP INDEX nomindex;

Comment (atm-gauthier) [reply htdocs/install/mysql/migration/17.0.0-18.0.0.sql]: I dit update

**Pull request merged**